src/evict/evict_lru.c

```diff
@@ -1726,6 +1726,15 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
     if (target_pages > remaining_slots)
         target_pages = remaining_slots;
 
+    /*
+     * Reduce the number of pages to be selected from btrees other than the history store (HS) if
+     * the cache pressure is high and HS content dominates the cache. Evicting unclean non-HS pages
+     * can generate even more HS content and will not help with the cache pressure, and will
+     * probably just amplify it further.
+     */
+    if (!WT_IS_HS(btree) && __wt_cache_hs_dirty(session))
+        target_pages = target_pages / 10;
```

Comment:
Please add a new statistic when this happens - it will be interesting to know that the cache is operating in this state.

Reply:
99266da